apiserver/src/main/java/org/dependencytrack/resources/v2/AdvisoriesResource.java

```diff
@@ -0,0 +1,375 @@
+
+/**
+ * API resources for advisories.
+ *
+ * @author Lawrence Dean
+ * @author Christian Banse
+ * @since 5.7.0
+ */
+@Path("/")
+public class AdvisoriesResource extends AbstractApiResource implements AdvisoriesApi {
+    private static final Logger LOGGER = Logger.getLogger(AdvisoriesResource.class);
+
+    @Override
+    @PermissionRequired(Permissions.Constants.VIEW_VULNERABILITY)
+    public Response listAdvisories(String format, String searchText) {
+        return inJdbiTransaction(getAlpineRequest(), handle -> {
+            // Normalize parameters: trim and treat empty as null so DAO SQL conditional
+            // behaves predictably
+            final String searchParam = (searchText == null || searchText.trim().isEmpty()) ? null
+                    : searchText.trim();
+            final String formatParam = (format == null || format.trim().isEmpty()) ? null : format.trim();
+
+            List<AdvisoryDao.AdvisoryDetailRow> advisoryRows = handle.attach(AdvisoryDao.class)
+                    .getAllAdvisories(formatParam, searchParam);
+            final long totalCount = handle.attach(AdvisoryDao.class)
+                    .getTotalAdvisories(formatParam, searchParam);
+
+            final List<ListAdvisoriesResponseItem> responseItems = advisoryRows.stream()
+                    .<ListAdvisoriesResponseItem>map(row -> ListAdvisoriesResponseItem.builder()
+                            .id(row.id())
+                            .title(row.title())
+                            .url(row.url())
+                            .seen(row.seen())
+                            .lastFetched(row.lastFetched().atOffset(ZoneOffset.UTC).toEpochSecond()*1000)
+                            .publisher(row.publisher())
+                            .name(row.name())
+                            .version(row.version())
+                            .format(row.format())
+                            .affectedComponents(row.affectedComponents())
+                            .affectedProjects(row.affectedProjects())
+                            .content(row.content())
+                            .build())
+                    .toList();
+
+            // Create a Page object for pagination metadata (no next page token at the moment)
+            final Page<ListAdvisoriesResponseItem> advisoriesPage = new Page<>(responseItems, null);
+
+            final ListAdvisoriesResponse response = ListAdvisoriesResponse.builder()
+                    .advisories(responseItems)
+                    .pagination(createPaginationMetadata(getUriInfo(), advisoriesPage))
+                    .build();
+
+            return Response.ok(response).header(TOTAL_COUNT_HEADER, totalCount).build();
+        });
+    }
+
+    @Override
+    @PermissionRequired(Permissions.Constants.VULNERABILITY_ANALYSIS_CREATE)
+    public Response uploadAdvisory(
+            @QueryParam("format") @NotNull String format,
+            @FormDataParam("file") InputStream _fileInputStream) {
```

Comment:
Not exactly sure why, but these annotations somehow cause the method definition in `AdvisoryApi` to be shadowed. And since this particular method doesn't have a `@Path`, Jersey doesn't register it as valid resource method at all.

Removing the annotations makes the tests pass.

```suggestion
            String format,
            InputStream _fileInputStream) {
```